Source/Assimil8or/Assimil8or.cpp

```diff
@@ -658,384 +657,45 @@ void Assimil8orPreset::parse (juce::StringArray presetLines)
                 switch (parseState)
                 {
                     case ParseState::ParsingGlobalSection: { jassertfalse; } break;
-                    case ParseState::ParsingPresetSection: { curPresetSection = {}; setParseState (ParseState::ParsingGlobalSection); } break;
-                    case ParseState::ParsingChannelSection: { curChannelSection = {}; setParseState (ParseState::ParsingPresetSection); } break;
-                    case ParseState::ParsingZoneSection: { curZoneSection = {}; setParseState (ParseState::ParsingChannelSection); } break;
+                    case ParseState::ParsingPresetSection:
+                    {
+                        curPresetSection = {};
+                        setParseState (ParseState::ParsingGlobalSection, &globalActions, "global");
+                    } break;
+                    case ParseState::ParsingChannelSection:
+                    {
+                        curChannelSection = {};
+                        setParseState (ParseState::ParsingPresetSection, &presetActions, "preset");
+                    } break;
+                    case ParseState::ParsingZoneSection: {
+                        curZoneSection = {};
+                        setParseState (ParseState::ParsingChannelSection, &channelActions, "channel");
+                    } break;
                     default: { jassertfalse; } break;
                 }
             }
         }
 
         LogParsing (juce::String (scopeDepth) + "-" + presetLine.trimStart ());
-        const auto key { presetLine.upToFirstOccurrenceOf (":", false, false).trim () };
-        const auto value { presetLine.fromFirstOccurrenceOf (":", false, false).trim () };
-
-        auto keyIs = [&key] (const juce::String& desiredKey)
-        {
-            return key.upToFirstOccurrenceOf (" ", false, false) == desiredKey;
-        };
-        auto getParameterIndex = [&key] ()
+        if (!presetLine.trim ().isEmpty ())
         {
-            return key.fromFirstOccurrenceOf (" ", false, false).getIntValue ();
-        };
-        switch (parseState)
-        {
-            case ParseState::ParsingGlobalSection:
-            {
-                if (keyIs (Section::PresetId))
-                {
-                    curPresetSection = presetProperties.getValueTree ();
-                    setParseState (ParseState::ParsingPresetSection);
-                }
-            }
-            break;
-            case ParseState::ParsingPresetSection:
+            key = presetLine.upToFirstOccurrenceOf (":", false, false).trim ();
+            value = presetLine.fromFirstOccurrenceOf (":", false, false).trim ();
+            const auto paramName = key.upToFirstOccurrenceOf (" ", false, false);
+            const auto action = curActions->find (paramName);
```

Comment:
for better scoping, move the `action` assignment into the start of the if statement `if (assignmentl; cond)`